### ISSUE TITLE
Lock-Free Data Structures: Request for a ARM-based server

### ISSUE THREAD

**Issue by junchangwang** [cncf/cluster#86]
Please fill out the details below to file a request for access to the CNCF Community Infrastructure Lab. Please note that access is targeted to people working on specific open source projects; this is not designed just to get your feet wet. The most important answer is the URL of the project you'll be working with. If you're looking to learn Kubernetes and related technologies, please try out [Katacoda](https://www.katacoda.com/courses/kubernetes/playground).

### First and Last Name
Junchang Wang

### Email
junchangwang@gmail.com

### Company/Organization
Nanjing University of Posts and Telecommunications

### Job Title
Researcher

### Project Title (i.e., summary of what do you want to do, not what is the name of the open source project you're working with)
Lock-free data structures.

### Briefly describe the project (i.e., what is the detail of what you're planning to do with these servers?)
I am interested in parallel computing and am working on helping improve Paul McKenney's classic textbook on parallel programming. This book is open-source, and I have submitted a dozen patches. However, all of the servers I can access are x86. So Paul kindly refers me to this website to apply access to ARM servers.

Besides, I have designed a few lock-free data structures (e.g., https://github.com/junchangwang/EQueue), and they are open-source under GPL license. Unfortunately, I never test them on ARM servers. 

### Is the code that you’re going to run 100% open source? If so, what is the URL or URLs where it is located? What is your association with that project?

Definity. Most of the projects I am working on are open source. For example:
Perfbook (contributer): https://mirrors.edge.kernel.org/pub/linux/kernel/people/paulmck/perfbook/perfbook.html
EQueue (owner): https://github.com/junchangwang/EQueue
ECCount (owner): https://github.com/junchangwang/eccount

### What kind of machines and how many do you expect to use (see: https://www.packet.net/bare-metal/)?
One ARM multi-core server.

### What OS and networking are you planning to use (see: https://help.packet.net/technical/infrastructure/supported-operating-systems)?
Ubuntu

### Any other relevant details we should know about?
No.

Thanks in advance!

**Comment by dankohn** [comment]
+1
--
Dan Kohn <dan@linuxfoundation.org>
Executive Director, Cloud Native Computing Foundation https://www.cncf.io
+1-415-233-1000 https://www.dankohn.com

**Comment by taylorwaggoner** [comment]
@junchangwang, I have sent you an invitation from Packet for your project.  Please let us know if you have questions.  Thanks!

**Comment by junchangwang** [comment]
Hi Taylor,

Thanks a lot for the help!

I wonder if there are any ARM (instead of Intel x86) based servers that I can use. Currently, in the Deploy on Demand Servers page, all of the servers I can choose are x86-based. What should I do to apply an ARM-based server?

Thanks,
--Junchang

**Comment by jacobsmith928** [comment]
Junchang,

We currently have limited supply of ARM systems while we move to new models.  Please contact us via help@packet.net so we can assist in getting you a proper machine.

**Comment by junchangwang** [comment]
Hello, Jacob,

Got it. Thanks!

**Comment by vielmetti** [comment]
Hi Junchang, a server has been reserved for your use from our reserve pool. It's lower spec than the standard system (32G of memory instead of the advertised 128G), 96-core Cavium ThunderX based.

**Comment by vielmetti** [comment]
This project has been completed, and has not seen activity since October 2021 when the last server was removed. I've removed the project from the CNCF CIL. If there is a need for any additional resources, please open a new issue. Thanks!
